generator_process/actions/control_net.py

```diff
@@ -265,6 +265,11 @@ def __call__(
                     num_warmup_steps = len(timesteps) - num_inference_steps * self.scheduler.order
                     with self.progress_bar(total=num_inference_steps) as progress_bar:
                         for i, t in enumerate(timesteps):
+                            # NOTE: Modified to support disabling CFG
+                            if (i / len(timesteps)) >= kwargs['cfg_end']:
+                                do_classifier_free_guidance = False
+                                prompt_embeds = prompt_embeds[None, 0]
+                                image = [i[None, 0] for i in image]
```

Comment:
This will likely need more changes for controlnet inpainting
```suggestion
                            if do_classifier_free_guidance and (i / len(timesteps)) >= kwargs['cfg_end']:
                                do_classifier_free_guidance = False
                                prompt_embeds = prompt_embeds[prompt_embeds.size(0) // 2:]
                                image = [i[None, 0] for i in image]
```

Reply:
Updated 👍